[Creationsv2] iP

Creationsv2 opened the pull request:
### CatBot
> "Your mind is for having idea,s not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

CatBot helps you manage your day to day items, freeing your mind of having to remember things. It's:

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is:

1. download it from [here](https://github.com/Creationsv2/ip)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😉 

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
___
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Creationsv2 committed:
Add increment Level-1.

Creationsv2 committed:
Added increment Level-1.

Creationsv2 committed:
Added increment Level-2

Creationsv2 committed:
Add increment Level-2.

Creationsv2 committed:
Merge remote-tracking branch 'origin/master'

Creationsv2 committed:
Add increment Level-3.

Creationsv2 committed:
Add increment Level-4

Creationsv2 committed:
Add A-TestUiTesting.

Creationsv2 committed:
Add increment Level-5.

Creationsv2 committed:
Add increment Level-6.

Creationsv2 committed:
Add A-Enums.

Creationsv2 committed:
Add javadocs.

Creationsv2 committed:
Add increment Level-7

Functionality to save list to, and load list from the hard disk.

Creationsv2 committed:
Add increment Level-8

Creationsv2 committed:
Merge branch 'branch-Level-7'

Creationsv2 committed:
Merge branch 'branch-Level-8'

Creationsv2 committed:
Fix bug in writing data

Creationsv2 committed:
Add increments A-MoreOOP & A-Packages

Creationsv2 committed:
Merge branch 'add-gradle-support' of https://github.com/Creationsv2/ip

Creationsv2 committed:
Add increment A-Gradle

Creationsv2 committed:
Add increment A-JUnit

Creationsv2 committed:
Add increment A-Jar

Creationsv2 committed:
Add increment A-JavaDoc

Creationsv2 committed:
Add increment A-CodingStandard

Creationsv2 committed:
Add increment Level-9

Creationsv2 committed:
Merge branch 'branch-A-JavaDoc'

Creationsv2 committed:
Merge branch 'branch-A-CodingStandard'

Creationsv2 committed:
Merge branch 'branch-Level-9'

Creationsv2 committed:
Add increment A-CheckStyle

Creationsv2 committed:
Add basic GUI functionality

anunayajoshi commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,107 @@
+    /** Duke.Storage component handling loading and writing tasks */
+    private Storage storage;
+    /** Task list with operations */
+    private TaskList tasks;
```
I like that you made the Tasklist plural.

anunayajoshi commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,58 @@
+     */
+    public Event(String description, String from, String to) throws DateTimeParseException {
+        super(description);
+        boolean saved = false;
```
Do you think that the boolean variable should be named isSaved instead?

anunayajoshi commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,68 @@
+    /** Stores the description of the task */
+    protected String description;
+    /** Stores the status of the task */
+    protected boolean isDone;
```
I like that the boolean variable is named with the prefix "is"

anunayajoshi commented on src/test/java/duke/task/EventTest.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    @Test
+    public void construtor_wrongDateFormat_exceptionThrown() {
```
small spelling mistake for constructor

anunayajoshi commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,124 @@
+ */
+public class Storage {
+    /** The file used to write/load data */
+    private File list;
```
perhaps the File variable could be named in a different way to make it more clear its a file object

anunayajoshi reviewed (commented):
Followed the coding standard very well, and put all classes in packages. Well done!

Creationsv2 committed:
Add improvements to GUI

Creationsv2 commented:
> Followed the coding standard very well, and put all classes in packages. Well done!

Thank you for the review, appreciate it 😄

Creationsv2 committed:
Update code quality

Creationsv2 committed:
Add increment A-Assertions

There are several methods that assume previous method's output is
correct.

Adding some assert statements will help to ensure program integrity.

Creationsv2 committed:
Update code quality.

There are several methods that have switch statements with repeated
code within case statements.

Moving those duplicated code out improves code quality and readability.

Creationsv2 committed:
Merge pull request #1 from Creationsv2/branch-A-Assertions

Add increment A-Assertions

Creationsv2 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Duke.java

Creationsv2 committed:
Merge pull request #3 from Creationsv2/branch-A-CodeQuality

Merge branch A-CodeQuality

Creationsv2 committed:
Add gradle.yml

Current project is not set up for github Workflow.

Setting up Continuous Integration with GitHub Actions will enable
automation of workflow.

Added gradle.yml to set up CI.

Creationsv2 committed:
Add extension C-Update

Current Catbot does not support updating descriptions of tasks if one
were to make a typo or wishes to change the description. He/she will
have to delete and add it back.

Update Catbot with a new command update to support such updates.

Creationsv2 committed:
Merge branch 'branch-C-Update'

Creationsv2 committed:
Add personality to CatBot

Creationsv2 committed:
Add product screenshot

Creationsv2 committed:
Update README.md

Creationsv2 committed:
Update README.md to include QuickStart & User Guide

Creationsv2 committed:
Update README.md with FAQ & Command Summary

Creationsv2 committed:
Update README.md to fix errors.

Creationsv2 committed:
Update Readme.md to fix errors.

Creationsv2 committed:
Update README.md to include release

Creationsv2 committed:
Fix bug in todo task

Creationsv2 committed:
Update README.md to include find command

Creationsv2 committed:
Update README.md

Creationsv2 committed:
Fix bug in deadline and event
